collectors/python.d.plugin/tomcat/tomcat.chart.py

```diff
@@ -96,6 +96,7 @@ def __init__(self, configuration=None, name=None):
         self.definitions = CHARTS
         self.url = self.configuration.get('url', 'http://127.0.0.1:8080/manager/status?XML=true')
         self.connector_name = self.configuration.get('connector_name', None)
+        self.single_quote_regex = re.compile(r"='([^']+)'([^']+)''")
```

Comment:
Python `re` package is thread safe, make it global var and just use in `get_data`. There is no reason to have copy of regex for in every instance

Reply:
Well you've made me realize a whole lot of my other projects code is not optimized !